common/persistence/dataInterfaces.go

```diff
@@ -1300,18 +1306,24 @@ type (
 		BranchToken []byte
 		// true means the fork is success, will update the flag, otherwise will delete the new branch
 		Success bool
+		// The shard to update history branch data
+		ShardID int
```

Comment:
ShardID should be pointer to make sure it is not a default value. Since zero is also a valid shardID for us.

Reply:
Will update it

Reply:
This change involves many changes in different place. Could you take another look?